assets/js/pages/Users/UsersPage.test.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+
+import MockAdapter from 'axios-mock-adapter';
+import { toast } from 'react-hot-toast';
+
+import { networkClient } from '@lib/network';
+import { screen, act } from '@testing-library/react';
+import { userEvent } from '@testing-library/user-event';
+import { adminUser, userFactory } from '@lib/test-utils/factories/users';
+import { renderWithRouter } from '@lib/test-utils';
+
+import UsersPage, {
+  SUCCESS_DELETE_MESSAGE,
+  ERROR_DELETE_MESSAGE,
+  ERROR_FETCH_MESSAGE,
+} from './UsersPage';
+
+const axiosMock = new MockAdapter(networkClient);
+
+jest.mock('react-hot-toast', () => ({
+  toast: {
+    success: jest.fn(),
+    error: jest.fn(),
+  },
+}));
+
+describe('UsersPage', () => {
+  afterEach(() => {
+    axiosMock.reset();
+    jest.clearAllMocks();
+  });
+
+  it('should render table without data', async () => {
+    axiosMock.onGet('/api/v1/users').reply(200, []);
+    await act(async () => {
+      renderWithRouter(<UsersPage />);
+    });
+    expect(await screen.getByText('No data available')).toBeVisible();
+  });
+
+  it('should render table with users', async () => {
+    const admin = adminUser.build();
+    const user = userFactory.build();
+
+    axiosMock.onGet('/api/v1/users').reply(200, [admin, user]);
+
+    await act(async () => {
+      renderWithRouter(<UsersPage />);
+    });
+
+    expect(await screen.getByText(admin.username)).toBeInTheDocument();
+    expect(await screen.getByText(user.username)).toBeInTheDocument();
+  });
+
+  it('should render toast with failing message when fetching users failed', async () => {
+    axiosMock.onGet('/api/v1/users').reply(404);
+    await act(async () => {
+      renderWithRouter(<UsersPage />);
+    });
+    expect(toast.error).toHaveBeenCalledWith(ERROR_FETCH_MESSAGE);
+  });
+
+  it('should render toast with success message when deleting was successfully', async () => {
+    const admin = adminUser.build();
+    const user = userFactory.build();
+
+    axiosMock.onGet('/api/v1/users').reply(200, [admin, user]);
+    axiosMock.onDelete(`/api/v1/users/${user.id}`).reply(204);
+
+    await act(async () => {
+      renderWithRouter(<UsersPage />);
+    });
+
+    let deleteButtons = screen.getAllByText('Delete');
```

Comment:
Why this `let` and a next reassignment?
It doesn't look like needed.
If you need to get the `Delete` button from the modal, use other `const` as they are different things